api/synapse-api-rest-imperative/src/main/java/io/americanexpress/synapse/api/rest/imperative/controller/helpers/PolyResponseEntityCreator.java

```diff
@@ -14,52 +14,51 @@
 package io.americanexpress.synapse.api.rest.imperative.controller.helpers;
 
 import io.americanexpress.synapse.service.imperative.model.BaseServiceResponse;
-import org.springframework.data.domain.Page;
+import io.americanexpress.synapse.service.imperative.model.PageResponse;
+import java.util.List;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.ResponseEntity.BodyBuilder;
+import org.springframework.http.ResponseEntity.HeadersBuilder;
 import org.springframework.util.CollectionUtils;
-import jakarta.servlet.http.HttpServletResponse;
-import java.util.List;
 
 /**
  * {@code PolyResponseEntityCreator} creates ResponseEntity for poly responses.
  * @param <O> a response extending {@link BaseServiceResponse}
  *
  * @author Francois Gutt
  */
-public class PolyResponseEntityCreator<O extends BaseServiceResponse> {
+public class PolyResponseEntityCreator {
 
     /**
      * Creates a Poly ResponseEntity with pagination.
      * @param page will be used for pagination.
-     * @param httpServletResponse response from the service.
      * @return ResponseEntity that will have {@link BaseServiceResponse}
      */
-    public static <O extends BaseServiceResponse> ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServletResponse) {
+    public static <O extends BaseServiceResponse> ResponseEntity<List<O>> create(PageResponse<O> page) {
         final ResponseEntity<List<O>> responseEntity;
         List<O> pageContent = null;
         if (page != null) {
-            pageContent = page.getContent();
+            pageContent = page.getResponsesForPage();
         }
         if (page == null || CollectionUtils.isEmpty(pageContent)) {
             responseEntity = new ResponseEntity<>(HttpStatus.NO_CONTENT);
         } else {
-            setHeadersInResponse(page, httpServletResponse);
-            responseEntity = new ResponseEntity<>(pageContent, HttpStatus.OK);
+            BodyBuilder bodyBuilder = ResponseEntity.ok();
```

Comment:
💪

Reply:
This wasnt working before, its working now :D